maple2latex/src/main.py

```diff
@@ -42,46 +40,65 @@ def obtain_formulae(self):
         return [MapleEquation(piece.split("\n")) for piece in contents.split("create(")
                 if "):" in piece or ");" in piece]
 
-    def convert_formulae(self):
-        return '\n\n'.join([make_equation(copy.copy(formula)) for formula in self.formulae])
-
     def __str__(self):
         return "MapleFile of " + self.filename
 
 
-def translate_file(filename):
-    # type: (str)
-    """Translates all the formulae in a file."""
-
-    with open("out/test.tex", "w") as test, open("out/primer") as primer:
-        test.write(primer.read() + MapleFile(filename).convert_formulae() + "\n\\end{document}")
-
-
-def translate_directories():
+def translate_files(root_directory):
     # type: ()
-    """Generates and writes the results of traversing the root directory and translating all Maple files."""
+    """Generates and writes the results of traversing the root directory, and translating all files in FILES."""
 
-    root_depth = len(ROOT.split("/"))
+    root_depth = len(root_directory.split("/"))
 
-    dirs = [d for d in os.walk(ROOT) if d[0].count("/") == root_depth]
+    dirs = [d for d in os.walk(root_directory) if d[0].count("/") == root_depth]
 
-    result = ""
+    sections = dict()
```

Comment:
Instead of `dict()`, simply declare it as `sections = {}`. `{}` is both more concise and approximately 3 times faster in Python (`timeit.timeit("{}") => 0.06282591819763184` vs. `timeit.timeit("dict()") => 0.1757810115814209`).